eval.ml

```diff
@@ -8,7 +8,8 @@ type trace = value list
 type environment = binding Env.t
 type heap = heap_value Heap.t
 
-type status = Running | Stopped
+type status = Running | Stopped of value option
```

Comment:
I like calling it `Result` instead of `Stopped`, but I think it's nice to be able to return without a value. (So I'm leaning towards a rename but keeping the `option`.)